app/experimenter/kinto/tasks.py

```diff
@@ -95,12 +99,12 @@ def nimbus_check_kinto_push_queue():
             generate_nimbus_changelog(
                 experiment,
                 get_kinto_user(),
-                message=f'Rejected: {rejected_collection_data["last_reviewer_comment"]}',
+                message=f'Rejected: {collection_data["last_reviewer_comment"]}',
             )
 
             kinto_client.rollback_changes()
 
-        if kinto_client.has_pending_review():
+        if collection_data["status"] == KINTO_REVIEW_STATUS:
```

Comment:
If you're doing the check explicitly here can't you just get rid of the `has_pending_review` method altogether?

Reply:
If we also want to update [this](https://github.com/mozilla/experimenter/blob/4b149db457b3b8eaedb151d1d1755cf8f37782f6/app/experimenter/kinto/tasks.py#L294) then yes

Reply:
Oh I seeeeeee.... the method is used elsewhere.... but if we try to use it here then it makes another call.  So we'd either have to remove the method and do the check explitily in the other spot the way we're doing here, or change this to cache the collection data on the client at init time, which is probably reasonable since within a single invocation of a task if it shifts during the middle of the task invocation we'd want the task to just fail hard and start again on its next invocation.

Which is basically what we're doing in the task you're updating, but then we kind of have two modes of using the client, either calling methods on it results in multiple collections calls, which I think we basically never want, or you explicitly pull in the collection data once and then just interact with it directly.

So that might be a better solution here, and then we can encapsulate all those status checks without having to expose the internals to the callers, and don't leave it up to the caller to manage how many calls get made to the collection.  What do you think?

Reply:
Sorry, just want to confirm: you’re saying we _should_ cache the collection data on init, and then update those methods to just access the cached data? And then leave the calls in the tasks untouched?

Reply:
Yeah I think that seems right?  Then within a specific task it looks like


```py
client = KintoClient(collection=, bucket=)
if client.has_some_state():
    client.do_some_thing()
```

and you know you're only interacting with the snapshotted state at the time the client was initialized, and then we can hide things like the kinto status checks entirely within the client module.  Right?

Reply:
Yeah I like that better than inspecting the raw collection data in the tasks. Lets the client do it’s job.